### DISCUSSION TITLE
go-eCharger bonjour discovery with mDNS-DS

### DISCUSSION THREAD

**Discussion by oh2th** [oh2th-homey/com.go-e.charger#7] [General]
To verify if your go-eCharger is automatically discoverable on your network, you can use [Bonjour Browser on Windows](https://hobbyistsoftware.com/bonjourbrowser) and [Discovery on macOS](https://apps.apple.com/fi/app/discovery-dns-sd-browser/id1381004916?mt=12). On the CLI in MacOS or Linux there is dns-sd tool.

Find the `_http._tcp` service  entry for your go-eCharger and there should be some information; `manufacturer`, `devicetype`, `serial` and `friendly_name` present for the discovery to work properly.

## Using Bonjour for Windows

In this example the go-eCharger is found with it's friendly name as named in the go-eCharger's mobile app.
![go-eCharger-http-bonjour](https://user-images.githubusercontent.com/59442255/209474763-47b7da3a-0fbb-42e2-a0cf-5c44c6f24939.png)

## Using Discovery on MacOS or iOS

Same should be found on macOS with [Discovery available from App Store](https://apps.apple.com/fi/app/discovery-dns-sd-browser/id1381004916?mt=12)

<img width="671" alt="Screenshot 2022-12-25 at 18 08 12" src="https://user-images.githubusercontent.com/59442255/209474978-b6dca062-ef77-433b-9986-093bff8bb15d.png">

## DNS-SD on the command line

First browse for the service instance and get the Instance Name:
```
dns-sd -t 1 -B
```
```
Browsing for _http._tcp
DATE: ---Wed 08 Feb 2023---
 6:10:32.990  ...STARTING...
Timestamp     A/R    Flags  if Domain               Service Type         Instance Name
 6:10:32.994  Add        3   6 local.               _http._tcp.          OctoPrint instance on octopi
 6:10:32.994  Add        3   6 local.               _http._tcp.          go-eCharger
 6:10:32.994  Add        3   6 local.               _http._tcp.          shellyem3-C45BBE6ADCCF
 6:10:32.994  Add        3   6 local.               _http._tcp.          shellyem3-8CAAB561F636
 6:10:32.994  Add        3   6 local.               _http._tcp.          shellyswitch25-98CDAC1DFD9F
 6:10:32.994  Add        3   6 local.               _http._tcp.          ShellyPro2PM-EC62608DEE44
```

To check the services on the instance name use:

```
dns-sd -t 1 -L  go-eCharger _http._tcp local
```
```
Lookup go-eCharger._http._tcp.local
DATE: ---Wed 08 Feb 2023---
 7:05:27.221  ...STARTING...
 7:05:27.223  go-eCharger._http._tcp.local. can be reached at go-echarger_201397.local.:80 (interface 6)
 protocol=2 version=055.0 devicetype=go-eCharger_V4 manufacturer=go-e friendly_name=go-eCharger serial=201397
```

**Comment by webmilesking** [comment]
There is the new go-e Charger PRO wallbox out wich needs to be added to your homey service: 
https://shop.go-e.com/en/wallbox/go-e-charger-pro-cable/
https://homey.app/de-de/app/com.go-e.charger/go-e-Charger/

Most important usecase is "SMART Charging" with own production from PV. 

If homey detects a home to grid power transfer (more solar energy production than usage) > then START charging wallbox go-e-Charger
If homey detects a home to grid power transfer more than xy watt>  then charge low / medium / high power

So the car will charge based only on available own add-on power production from home,  which would be transferred to grid instead.

![CH-PRO-CABLE-001_0_de-DE](https://github.com/user-attachments/assets/bd465309-99f2-4ad1-9c88-b5227974aa18)

**Reply by oh2th** [reply]
Not related on mDNS. Please stay on topic. Start a new discussion and this will be deleted.
